hypervisor/src/kvm/x86_64/mod.rs

```diff
@@ -16,3 +20,21 @@ pub use {
     kvm_bindings::kvm_xcrs as ExtendedControlRegisters, kvm_bindings::kvm_xsave as Xsave,
     kvm_bindings::CpuId, kvm_bindings::Msrs as MsrEntries,
 };
+
+pub const KVM_TSS_ADDRESS: GuestAddress = GuestAddress(0xfffb_d000);
+
+///
+/// Check KVM extension for Linux
+///
+pub fn check_required_kvm_extensions(kvm: &Kvm) -> KvmResult<()> {
+    if !kvm.check_extension(Cap::SignalMsi) {
+        return Err(KvmError::CapabilityMissing(Cap::SignalMsi));
+    }
+    if !kvm.check_extension(Cap::TscDeadlineTimer) {
+        return Err(KvmError::CapabilityMissing(Cap::TscDeadlineTimer));
+    }
+    if !kvm.check_extension(Cap::SplitIrqchip) {
+        return Err(KvmError::CapabilityMissing(Cap::SplitIrqchip));
+    }
+    Ok(())
```

Comment:
This is very Cloud-Hypervisor specific. I mean the requirement for having these KVM caps supported is not something that all VMM might need, so it'd be more appropriate to let the VMM check them.

Reply:
In this case we need to have a way to return this kvm object reference back to vmm.